api/lib/opentelemetry/trace/trace_flags.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    # TraceFlags contain details about the trace. Unlike Tracestate values,
+    # TraceFlags are present in all traces. Currently, the only TraceFlag is a
+    # boolean {sampled?} {https://www.w3.org/TR/trace-context/#trace-flags flag}.
+    class TraceFlags
+      class << self
+        private :new # rubocop:disable Style/AccessModifierDeclarations
+
+        # Returns a newly created {TraceFlags} with the specified flags.
+        #
+        # @param [Integer] flags 8-bit byte of bit flags
+        # @raise [ArgumentError] If flags is not an 8-bit byte
+        # @return [TraceFlags]
+        def from_byte(flags)
+          raise ArgumentError, 'flags must be an 8-bit byte' unless (0..255).include?(flags)
```

Comment:
Is it more efficient to `unless flags & ~0xff == 0`?

Reply:
The bitwise check is a clear winner:

```ruby
require 'benchmark/ipsa'

num = 144

Benchmark.ipsa do |x|
  x.report 'range check' do
    (0..255).include?(num)
  end
  x.report 'bitwise check' do
    num & ~0xFF == 0
  end
  x.compare!
end
```

Results:

```
Allocations -------------------------------------
         range check       0/0  alloc/ret        0/0  strings/ret
       bitwise check       0/0  alloc/ret        0/0  strings/ret
Warming up --------------------------------------
         range check   180.625k i/100ms
       bitwise check   205.490k i/100ms
Calculating -------------------------------------
         range check      7.869M (± 4.7%) i/s -     39.376M
       bitwise check     15.487M (± 2.7%) i/s -     77.470M

Comparison:
       bitwise check: 15486958.1 i/s
         range check:  7868744.9 i/s - 1.97x slower
```